### PR TITLE
feat: Added blurhash to image metadata

### LINKED ISSUE #12351: Add blurhash to image metadata
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are interested into the same feature.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

---

## Feature request

**Is your feature request related to a problem? Please describe.**
For showing a preview of an image while it is still loading a blurhash can be used. It is not currently exposed in the metadata though. This was supposed to happen with https://github.com/nextcloud/spreed/issues/10844, but it wasn't added in https://github.com/nextcloud/spreed/pull/11093.

### PR DESCRIPTION
**Pull request by skalidindi53**
### ☑️ Resolves

* Fix #12351 

### 🚧 Tasks

- [x] Added blurhash to image metadata

### 🏁 Checklist

- [x] ⛑️ Tests (unit and/or integration) are included or not possible
- [ ] 📘 API documentation in `docs/` has been updated or is not required
- [ ] 🔖 Capability is added or not needed

### CODE REVIEW

File: lib/Chat/Parser/SystemMessage.php [hunk 1]
```diff
@@ -817,6 +817,10 @@ protected function getFileFromShare(Room $room, ?Participant $participant, strin
 					$data['width'] = (string) $sizeMetadata['width'];
 					$data['height'] = (string) $sizeMetadata['height'];
 				}
+
+				if (isset($sizeMetadata['blurhash'])) {
+					$data['blurhash'] = $sizeMetadata['blurhash'];
```
Comment (nickvergessen) [inline lib/Chat/Parser/SystemMessage.php:822]: Hmm we should add it to https://github.com/nextcloud/server/blob/master/lib/public/RichObjectStrings/Definitions.php#L294 and https://github.com/nextcloud/spreed/blob/main/lib/ResponseDefinitions.php#L93 I guess

**Comment by nickvergessen** [comment]
@SystemKeeper can/want to give this a try?

**Review by nickvergessen — APPROVED** [review]

**Comment by SystemKeeper** [comment]
> @SystemKeeper can/want to give this a try?

The PR works fine for me and correctly returns the blurhash. 

Would it make sense to rename size-related methods/variables like `getMetadataPhotosSizeForFileId` to `getImageMetadataForFileId` or the like to make it clearer that this is not only about size?

One thing I noticed is that blurhash generation has since been moved to a background job: https://github.com/nextcloud/server/blob/560282a47bf3f9dca4653f89d530a6e1dd144f5f/lib/private/Blurhash/Listener/GenerateBlurhashMetadata.php#L57-L61

So when we upload images, a blurhash will only be available after the next background run, which might limit the usefulness a bit...

Example from ios:

**Review by SystemKeeper — APPROVED** [review]

**Comment by nickvergessen** [comment]
> One thing I noticed is that blurhash generation has since been moved to a background job

Hmpf, makes sense performancewise, but yeah kind of kills the feature

**Pull request merged**